src/token-provider.js

```diff
@@ -32,6 +31,9 @@ export default class TokenProvider {
   }
 
   makeAuthRequest() {
+    if (!this.authEndpoint) {
+      throw new TypeError("Please configure an authEndpoint to gain access to private feeds. (See http://docs.pusher.com/feeds/concepts/private-feeds/)");
```

Comment:
I don't think a `TypeError` is appropriate here. How about defining a custom one like `AuthEndpointNotInitializedError`?

Reply:
It looks like custom errors are surprisingly fiddly in javascript, I'm not sure I fully understand how they'll behave in older browsers and stuff. I can give it a shot though and maybe chat to @hph about it. (I think it needs some babel extensions and stuff at the very least)

Reply:
@WillSewell In fact, the code base is full of generic errors. So introducing custom errors is probably outside of the scope of this PR?

Reply:
By "generic error", do you mean something like `throw "you did this wrong"`? i.e. no specific type. I'm happy for you to do whatever the current convention is, I just don't think `TypeError` is appropriate.

Reply:
mostly `throw TypeError("you did this wrong")`

I might be wrong about this, which is why I'd like to chat to hawk about it, but I think `TypeError` is used for a wider class of things than you would expect in JS. e.g. stuff like https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/More_arguments_needed

Basically when any variable is now of the type or format you want, or is missing. If you squint then this _is_ a type error, because `this.authEndpoint` has type `undefined` instead of type `string`, which is what we want. In fact I should probably just change the error check to actually check the type, and then we cover ourselves against people passing truthy non strings too...

Reply:
Hmm I guess it makes sense for function arguments because you are validating the value is of the type you would expect. It just feels strange doing the validation on the object the method is attached to, although you could view the object as the first argument to the function so I guess it kinda makes sense in that respect.

Feel free to leave it as-is if it's done elsewhere and is common practice in JS.